packages/@aws-cdk/aws-cloudwatch/lib/metric.ts

```diff
@@ -388,16 +417,22 @@ export class Metric implements IMetric {
    */
   private dimensionsAsList(): Dimension[] {
     const dims = this.dimensions;
+    const dimsMap = this.dimensionsMap;
```

Comment:
If you populate `this.dimensions` from `props.dimensionsMap`, like I suggested above, you won't have to change anything here!